javascript/javascript1/week4/voice_assistant.js

```diff
@@ -136,22 +156,58 @@ function calculator(a, b, operation) {
 }
 
 function calculateResult(input) {
-  const regex = /what is (\d+)\s*([+\-*\/])\s*(\d+)\?/i;
-  const match = input.match(regex);
+  const match = input.match(patterns.mathPattern);
   if (match && match.length === 4) {
-    const num1 = parseInt(match[1]);
+    const num1 = parseInt(match[1]); //extracts numbers and operation from a string
     const operation = match[2];
     const num2 = parseInt(match[3]);
     return calculator(num1, num2, operation);
   } else {
     return "Invalid input.";
   }
 }
-//console.log(greetings("Hello, my name is Tanya"));
-console.log(addToDo("Add dishes to my to do list."));
+
+function setTimer(input) {
+  const match = input.match(patterns.timerPattern);
+  if (match && match.length === 2) {
+    const minutes = parseInt(match[1]);
+    setTimeout(function () {
+      console.log("Time is up!");
+    }, minutes * 60 * 1000);
```

Comment:
Great and works as expected. You have the fundamentals down pat but in the future of the your Javascript journey you will notice how setting timeouts can give you wonky behaviour (especially in React) then multiple setTimeouts will run on refresh.

Here you will become familiar with the importance clearing your setTimeouts (clearTimeout). It's definitely beyond what we have learnt but you are far ahead and may want some extra reading then read this [article on setTimout & clearTimeout](https://alonewolf24.medium.com/mastering-javascripts-settimeout-and-cleartimeout-functions-7eb63a369175)

Reply:
Perfect! Thank you for sharing!